assets.src/src/redmine-tracky/controllers/timer-controller.ts

```diff
@@ -56,18 +55,28 @@ export default class extends Controller {
         }
     }
 
-    private timeDifference(): string {
-        const startDateTime = this.convertToDateTime((this.startTarget as HTMLInputElement).value);
-        const endDateTime = this.convertToDateTime((this.endTarget as HTMLInputElement).value);
+    private timeDiffToString(timeDiff: TimeDiff) {
+        const sign = (timeDiff.minutes < 0 || timeDiff.seconds < 0) ? '-' : '';
+    
+        return sign + ['hours', 'minutes', 'seconds']
+            .map((v) => timeDiff[v as keyof TimeDiff])
```

Comment:
i'm fine with your approach - you could avoid this cast with my suggestion. that's also why i switched to Duration. But there is no right or wrong. if you prefer you approach fine for me.